source/Nevermore/Advanced/WriteTransaction.cs

```diff
@@ -282,20 +282,24 @@ public void Commit()
         {
             if (Transaction is null)
                 throw new InvalidOperationException("There is no current transaction, call Open/OpenAsync to start a transaction");
+
             if (!configuration.AllowSynchronousOperations)
                 throw new SynchronousOperationsDisabledException();
 
             configuration.Hooks.BeforeCommit(this);
             Transaction.Commit();
+            ExecutedCommands.Clear();
```

Comment:
Once the transaction is committed we need to clear the list of executed commands.